dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Extensions/KernelChatGptPluginExtensions.cs

```diff
@@ -0,0 +1,150 @@
+                //using DefaultHttpRetryHandler retryHandler = new DefaultHttpRetryHandler(
+                //  config: new HttpRetryConfig() { MaxRetryCount = 3 },
+                //  log: null);
+
+                //using HttpClient client = new HttpClient(retryHandler, false);
+                using HttpClient client = new HttpClient();
+
+                response = await client.GetAsync(url);
+            }
+            else
+            {
+                response = await httpClient.GetAsync(url);
+            }
+            response.EnsureSuccessStatusCode();
+
+            string gptPluginJson = await response.Content.ReadAsStringAsync();
+            string? openApiUrl = ParseOpenApiUrl(gptPluginJson);
+
+            return await kernel.ImportOpenApiSkillFromUrlAsync(skillName, new Uri(openApiUrl), httpClient);
+        }
+        finally
+        {
+            if (response != null)
+            {
+                response.Dispose();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Imports ChatGPT plugin from assembly resource.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="httpClient">Optional HttpClient to use for the request.</param>
+    /// <returns>A list of all the semantic functions representing the skill.</returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportChatGptPluginSkillFromResourceAsync(
+        this IKernel kernel, string skillName, HttpClient? httpClient = null)
+    {
+        Verify.ValidSkillName(skillName);
+
+        var type = typeof(SkillResourceNames);
+
+        var resourceName = $"{skillName}.ai-plugin.json";
+
+        var stream = type.Assembly.GetManifestResourceStream(type, resourceName);
+        if (stream == null)
+        {
+            throw new MissingManifestResourceException($"Unable to load OpenApi skill from assembly resource '{resourceName}'.");
+        }
+
+        using StreamReader reader = new StreamReader(stream);
+        string gptPluginJson = await reader.ReadToEndAsync();
+
+        string? openApiUrl = ParseOpenApiUrl(gptPluginJson);
+
+        return await kernel.ImportOpenApiSkillFromUrlAsync(skillName, new Uri(openApiUrl), httpClient);
+    }
+
+    /// <summary>
+    /// Imports ChatGPT plugin from a directory.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="parentDirectory">Directory containing the skill directory.</param>
+    /// <param name="skillDirectoryName">Name of the directory containing the selected skill.</param>
+    /// <param name="httpClient">Optional HttpClient to use for the request.</param>
+    /// <returns>A list of all the semantic functions representing the skill.</returns>
+    public static IDictionary<string, ISKFunction> ImportChatGptPluginSkillSkillFromDirectory(
```

Comment:
Nit: I'd probably replace this one with "...FromFile" and just require the path to the json file itself. That way, it could be named anything, and there could be various versions ("ai-plugin-test.json") and you wouldn't have to put them all in different directories just to load one specific file.

Reply:
This is the pattern with other extensions, but I'll add a "from file" one as well.